### DISCUSSION TITLE
Partial and Lazy hydration will be on Vue 3.3 core

### DISCUSSION THREAD

**Discussion by qnp** [ElMassimo/iles#55] [General]
Hi @ElMassimo,
First of all, thanks for the amazing work. 

Recently I came across [this Nuxt Nation 2021 conference](https://youtu.be/KsIN33ANybM?t=923) with Evan Yu answering questions. At min 15:23 he is mentioning the work being carried out at the moment by the core team, and planned to be shipped on Vue version 3.3, on Partial and Lazy hydration.

Does this mean that Îles framework will be obsolete for Vue ?

Best regards

**Comment by ElMassimo** [comment]
@qnp Hi François!

Having support for partial hydration in Vue core would simplify the implementation of îles, which is great for me as a maintainer 😃

îles would not become obsolete, as it will still provide many features that are useful when generating static sites, such as MDX support, layouts, support for other frameworks that have smaller runtimes, and [other useful features](https://iles-docs.netlify.app/guide/development#pages).

**Reply by qnp** [reply]
Thank you for your answer. It's interesting. Do you have an estimation of when Îles would be in stable release ?

**Reply by ElMassimo** [reply]
The release of 1.0 will depend on the implementation of certain roadmap items which I intend to share soon, which include but are not limited to:

- [x] Vitepress-compatibility mode (hrefs include `.html`)
- [x] Markdown excepts (@islands/excerpt)
- [x] Official RSS module (@islands/feed)
- [x] Collection helpers (`useDocuments`)
- [x] Official images module (@islands/images)
- [x] [Official PWA module](https://iles-docs.netlify.app/guide/pwa)
- `useAsyncData` and `useFetch` helpers (can already be done, just syntax sugar because it's very common)
- Rendering dynamically fetched markdown using the MDX compiler ([can already be done](https://github.com/YuukanOO/iles-markdown-sandbox) but requires manual setup, and is not beginner friendly)
- Official i18n module ([can already be done](https://github.com/ElMassimo/iles/discussions/6) using `enhanceApp`)

---

Most of these items are related to the DX of generating a static site, which I would like to improve to the point where it's enjoyable for everyone, from beginner devs to experienced Vue users.

The parts of the API that are related to layouts, routing, components, and partial hydration can be already considered stable. Any changes in these will be backwards compatible.

**Reply by TechAkayy** [reply]
Can we refer to this checklist to know how we are tracking towards stable release? Wondering if this list is actively referred to & maintained. Thanks Massimo.

**Reply by ElMassimo** [reply]
Hi @TechAkayy! I've been keeping this list up to date. It would be nice to create more visible roadmap in the website, but that will have to wait a few months (too busy at the moment).
